### DISCUSSION TITLE
Filter top level with the deep level filter

### DISCUSSION THREAD

**Discussion by Disura-Randunu** [directus/directus#8054] [Archive]
When querying, is it possible to exclude items at the top level which pass the deep level filter?

**Comment by rijkvanzanten** [comment]
You can use a nested notation in the filter structure, for example:

Get all articles where the author's name is Rijk

```
GET /items/articles
  ?filter[author][name][_eq]=Rijk
```

Get all articles, but only fetch the nested comments that are posted by Rijk

```
GET /items/articles
  ?fields=*.*
  &deep[comments][_filter][author][name][_eq]=Rijk
```

**Reply by Disura-Randunu** [reply]
If it's no trouble, can you show how can it be done through the JavaScript SDK?

This is what I did. It just make the deep level item an empty array. Doesn't exclude the top level elements.What could I be doing wrong?

```js
{
        fields:'*.*', 
        deep:{
            relationship_type:{
                _filter:{
                    relationship_types_id:{
                        relationship_name:{
                            _eq:'project_gitlab'
                        }
                    }
                } 
            }
        }
}
```

**Reply by rijkvanzanten** [reply]
> Doesn't exclude the top level elements.

This checks out, as you aren't using `filter` on the parent level, try:

```js
{
  fields: "*.*",
  filter: {
    relationship_type: {
      relationship_types_id: {
        relationship_name: {
          _eq: 'project_gitlab'
        }
      }
    }
  }
}
```

(I obviously don't know what your schema looks like, so the actual names might be different)

**Reply by Disura-Randunu** [reply]
It works!!!!!! Thank you so much!!
